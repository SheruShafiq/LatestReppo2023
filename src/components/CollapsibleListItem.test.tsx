import matchers from "@testing-library/jest-dom";
expect.extend(matchers);

import { render } from "@testing-library/react";
import {
  CollapsibleListItemProps,
  CollapsibleListItem,
} from "../components/CollapsibleListItem";
import { debug } from "vitest-preview";

describe("FAQListItem", () => {
  const defaultProps: CollapsibleListItemProps = {
    title: "Test question",
    content: "Test answer",
  };

  it("renders the question and answer correctly", async () => {
    const { getByText } = await render(
      <CollapsibleListItem {...defaultProps} />
    );

    expect(getByText(defaultProps.title)).toBeInTheDocument();
    expect(getByText(defaultProps.content)).toBeInTheDocument();
  });

  it("expands and collapses the accordion", async () => {
    const { getByText, getByRole } = await render(
      <CollapsibleListItem {...defaultProps} />
    );
    const summary = getByRole("button");
    expect(getByText(defaultProps.content)).not.toBeVisible();

    summary.click();
    expect(getByText(defaultProps.content)).not.toHaveStyle("display: none");

    debug();
    summary.click();
    expect(getByText(defaultProps.content)).not.toBeVisible();
    expect(screen).toMatchSnapshot();
  });
});
