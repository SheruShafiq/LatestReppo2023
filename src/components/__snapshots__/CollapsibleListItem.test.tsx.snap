// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FAQListItem > expands and collapses the accordion 1`] = `Screen {}`;
